cli/src/cmd/forge/script/cmd.rs

```diff
@@ -0,0 +1,202 @@
+use crate::{
+    cmd::{unwrap_contracts, Cmd, ScriptSequence},
+    utils::get_http_provider,
+};
+
+use ethers::{
+    abi::Abi,
+    prelude::{artifacts::CompactContractBytecode, ArtifactId, Middleware, Signer},
+    solc::utils::RuntimeOrHandle,
+    types::{transaction::eip2718::TypedTransaction, U256},
+};
+use forge::executor::opts::EvmOpts;
+
+use foundry_config::{figment::Figment, Config};
+
+use super::*;
+
+impl Cmd for ScriptArgs {
+    type Output = ();
+    fn run(self) -> eyre::Result<Self::Output> {
+        RuntimeOrHandle::new().block_on(self.run_script())
+    }
+}
+
+impl ScriptArgs {
+    pub async fn run_script(mut self) -> eyre::Result<()> {
+        let figment: Figment = From::from(&self);
+        let evm_opts = figment.extract::<EvmOpts>()?;
+        let mut script_config = ScriptConfig {
+            // dapptools compatibility
+            sender_nonce: U256::one(),
+            config: Config::from_provider(figment).sanitized(),
+            evm_opts,
+            called_function: None,
+        };
+
+        self.maybe_load_private_key(&mut script_config)?;
+
+        if let Some(fork_url) = script_config.evm_opts.fork_url.as_ref() {
+            script_config.sender_nonce =
+                foundry_utils::next_nonce(script_config.evm_opts.sender, fork_url, None)?
+        } else {
+            script_config.config.libraries = Default::default();
+        }
+
+        let BuildOutput {
+            project,
+            target,
+            contract,
+            highlevel_known_contracts,
+            predeploy_libraries,
+            known_contracts: default_known_contracts,
+            sources,
+        } = self.build(&script_config)?;
+
+        if self.resume {
+            let fork_url = self
+                .evm_opts
+                .fork_url
+                .as_ref()
+                .expect("You must provide an RPC URL (see --fork-url).")
+                .clone();
```

Comment:
can we do this first, so we fail early?

Reply:
what is first in this case? 
we actually only want to fail if the user is broadcasting or resuming anything, which might not always be the case...